### PR TITLE
Add Dataset Factory v0 durability gate

### LINKED ISSUE #65: Dataset Factory v1: land and reproduce the v0 implementation
Parent: #64

## Goal

Move the locally proven Dataset Factory v0 implementation onto a durable, reviewed branch and prove that its important workflows can be reconstructed from a clean checkout.

This is the first gate in Dataset Factory v1. No new modeling work should depend on local-only source files.

## Current evidence

The canonical checkout currently contains:

- 49 untracked files under `docs/dataset-factory/` and `packages/scripts/src/dataset-factory/`;
- modified Dataset Factory command wiring in the root and scripts-workspace `package.json` files;
- Dataset Factory status changes in `TASKS.md`;
- substantial quality-audit changes in `packages/scripts/src/autoresearch/image-quality-issues.ts`;
- large generated reports under `data/mtl_archives/reports/`, intentionally ignored by git;
- unrelated social-pipeline changes in the same dirty checkout;
- a local `main` that must not be used as an undifferentiated commit source.

The local source currently passes:

- `npm run typecheck --workspace=@mtl-archives/scripts`;
- JSON/JSONL parsing for Dataset Factory schemas and examples;
- `git diff --check`.

That proves syntax health, not durability or clean-clone reproducibility.

## Scope

- Create an isolated worktree from current `origin/main`.
- Inventory every v0 source, schema, documentation, command-wiring, and task-board change.
- Separate Dataset Factory changes from unrelated social/content work.
- Bring forward only the intended Dataset Factory surface.
- Add small deterministic fixtures for the major command families:
  - packet generation;
  - label validation/adjudication;
  - benchmark generation;
  - active-learning acquisition;
  - search/reranker evaluation;
  - quality-repair planning;
  - visual-family graph;
  - research enrichment;
  - reward-data generation.
- Add a machine-readable artifact registry for large ignored inputs/outputs.
- Record hashes, row counts, schema versions, creation commands, and storage locations without committing large generated data.
- Update the repo documentation required by `AGENTS.md`: `README.md`, `FORWIEL.md`, `TASKS.md`, `docs/architecture.md`, and `CLAUDE.md` where behavior or operator commands changed.

## Artifact policy

Tracked:

- source code;
- schemas;
- small fixtures;
- command documentation;
- artifact manifests and checksums;
- compact run reports and decisions.

Untracked or external:

- full image derivatives;
- large JSONL corpora;
- embeddings;
- model checkpoints;
- generated contact sheets;
- raw credentials or signed URLs.

Every external artifact used by a reproducible command must have a registry entry with:

- stable artifact ID;
- schema version;
- SHA-256;
- row/file count;
- source lineage;
- storage class/location;
- generating command;
- dependency artifact IDs;
- rights/attribution boundary;
- created timestamp.

## Acceptance criteria

- [ ] All intended Dataset Factory v0 source, schemas, and docs are tracked on a scoped branch.
- [ ] No unrelated social/content files are included in the Dataset Factory diff.
- [ ] Root and workspace command wiring is complete and internally consistent.
- [ ] A clean checkout can install dependencies and run Dataset Factory fixture tests.
- [ ] `npm run typecheck --workspace=@mtl-archives/scripts` passes.
- [ ] All tracked JSON/JSONL schemas, examples, and registries parse successfully.
- [ ] `git diff --check` passes.
- [ ] The artifact registry covers every ignored artifact required by a documented v0 workflow.
- [ ] A clean-clone smoke report records commands, outputs, hashes, and limitations.
- [ ] A separate reviewer checks correctness, scope, accidental secret inclusion, and reproducibility.
- [ ] The PR is merged before downstream v1 implementation begins.

## Stop conditions

Stop and report instead of merging if:

- an artifact required for reproduction cannot be located or identified;
- hashes/row counts disagree with the GitHub issue closeout evidence;
- a Dataset Factory file cannot be separated safely from unrelated dirty-worktree changes;
- fixture tests reveal nondeterminism or hidden credential requirements;
- secrets, private keys, tokens, or temporary signed URLs are found.

## Non-goals

- No production D1/R2/Vectorize writes.
- No Worker/site deployment.
- No paid GPU launch.
- No attempt to commit the full ignored report tree.
- No new label batch or model training.

### PR DESCRIPTION
**Pull request by zouantchaw**
## Summary

Closes #65.

- Restores Dataset Factory v0 docs/schemas and TypeScript entrypoints from canonical source evidence, with intentional #65 durability additions only.
- Adds a 27-entry machine-readable artifact registry plus JSON schema for ignored v0 artifacts, including stable IDs, SHA-256 digests, file/row counts where meaningful, lineage, storage/path class, generation commands, dependency IDs, rights boundaries, and timestamps.
- Adds tracked v0 smoke fixtures and `dataset-factory:smoke-v0` covering packets, labels/adjudication, benchmark, search/reranker evaluation, active learning, quality repair, visual family graph, research enrichment, search judgments, and reward data without copying generated report trees.
- Adds missing-artifact diagnostics and updates README/FORWIEL/TASKS/architecture/CLAUDE guidance.

## Verification

- `npm run dataset-factory:artifacts:check` -> 27 entries, registry valid.
- `npm run dataset-factory:artifacts:check -- --verify-files --artifact-root /Users/wiel/Development/mtl-archives-search` -> all registered ignored artifacts matched canonical SHA/count metadata.
- `npm run dataset-factory:smoke-v0` -> ok; key rows: review packets 6, batch packets 5, adjudicated labels 5, retrieval tasks 6, search baseline rows 18, reranker candidates 48, active-learning rows 5, quality repair rows 7, visual families 3, research packets 8, search judgments 10, reward signals 33.
- `npm run typecheck --workspace=@mtl-archives/scripts` -> pass.
- Dataset Factory JSON/JSONL/NDJSON parse -> 24 files, 81 parsed JSONL/NDJSON rows.
- `git diff --check` / staged `git diff --cached --check` -> pass.
- Clean clone smoke from `/tmp/mtl-issue65-clean.jFaWrK/repo` after `npm ci` -> artifact registry check, `dataset-factory:smoke-v0`, and scripts typecheck passed using tracked files/fixtures only.
- Diff audit -> no files over 1 MB, no `data/` generated report tree staged, no `.env`/credential files, no Cloudflare/D1/R2/Vectorize/social/API mutation paths staged.

## Notes / Residual Limits

- Clean clone `npm ci` warns that the current shell is Node `v22.22.0` while the repo declares `>=23.5.0`; the tested commands still pass under this shell.
- Fixture smoke validates contracts and command wiring with a local mock `/api/search`; it is not a live API ranking-quality proof.
- Full artifact hash verification still requires a populated local artifact root; large generated outputs remain ignored by design.

### CODE REVIEW

File: docs/dataset-factory/artifact-registry.v0.jsonl [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
...ent_digest":{"algorithm":"sha256","value":"e569177884857876133ab389a18b52d26206bbe5697def4177beb5599287230a","scope":"sorted_tree_manifest"},"counts":{"file_count":10,"byte_count":316795,"row_count":860},"source_lineage":{"description":"Initial MTL-CityMemory-Bench v0 artifact built from calibration labels.","source_artifact_ids":["dfv0_batch_001"],"source_urls":["https://github.com/zouantchaw/mtl-archives-search/issues/65"]},"storage":{"storage_class":"local_filesystem","path_class":"repo_ignored_local","locator":"data/mtl_archives/reports/dataset_factory_benchmark_v0"},"generation":{"command":"npm run dataset-factory:benchmark-v0","code_ref":"codex/65-dataset-factory-v0-durability"},"dependency_ids":["dfv0_batch_001"],"required_by":["dataset-factory:benchmark-v0:search"],"rights_boundary":{"license_id":"cc-by-4.0-derived","attribution":"Archives de la Ville de Montreal; generated Dataset Factory analysis by MTL Archives.","commercial_use_allowed":true,"notes":"Derived metadata/evaluation artifacts inherit source attribution requirements; generated labels remain review artifacts, not source facts."},"created_at":"2026-06-29T11:37:19.577Z","creation_time_basis":"latest_file_mtime"}
```
Comment (zouantchaw) [inline docs/dataset-factory/artifact-registry.v0.jsonl]: [P1] Make this artifact graph reconstructible. `dfv0_batch_001` depends on `dfv0_active_learning_v0`, which depends on `dfv0_benchmark_v0_initial`, which depends back on `dfv0_batch_001` (there is a second cycle through model-baseline/quality-gold). The recorded batch command also cannot reproduce the registered 53-file tree because that hash includes visual-review, OCR-review, and quality-model-review subtrees whose apply commands and human decision inputs are omitted. A clean checkout therefore has no valid generation order or byte-reproduction command. Split this broad directory into immutable phase artifacts (including the reviewed decision inputs), record exact commands/dependencies for each phase, and make the checker reject dependency cycles.

File: docs/dataset-factory/artifact-registry.v0.jsonl [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
...udit_v0","artifact_kind":"file","content_digest":{"algorithm":"sha256","value":"2673a86c8a43b4f026b8d60984a2d5e7eb398aefb8765529b86575a6531b4488","scope":"file_bytes"},"counts":{"file_count":1,"byte_count":720},"source_lineage":{"description":"Orientation audit report feeding conservative rotation review.","source_artifact_ids":["dfv0_manifest_clean"],"source_urls":["https://github.com/zouantchaw/mtl-archives-search/issues/65"]},"storage":{"storage_class":"local_filesystem","path_class":"repo_ignored_local","locator":"data/mtl_archives/reports/image_orientation_audit.json"},"generation":{"command":"npm run image-orientation:audit","code_ref":"codex/65-dataset-factory-v0-durability"},"dependency_ids":["dfv0_manifest_clean"],"required_by":["dataset-factory:quality-repair-v0"],"rights_boundary":{"license_id":"cc-by-4.0-derived","attribution":"Archives de la Ville de Montreal; generated Dataset Factory analysis by MTL Archives.","commercial_use_allowed":true,"notes":"Derived metadata/evaluation artifacts inherit source attribution requirements; generated labels remain review artifacts, not source facts."},"created_at":"2026-02-20T10:28:22.033Z","creation_time_basis":"filesystem_mtime"}
```
Comment (zouantchaw) [inline docs/dataset-factory/artifact-registry.v0.jsonl]: [P1] Wire the command recorded here. From the repository root, `npm run image-orientation:audit` exits 1 with `Missing script`; the command exists only in `packages/scripts/package.json`. Line 13 records the same unusable root command, and both artifacts feed `quality-repair-v0`. Add the root workspace forwarder or record the workspace-qualified command, then include a command-existence check.

File: packages/scripts/src/dataset-factory/check-artifact-registry.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+      'artifact-root': { type: 'string', default: MONOREPO_ROOT },
+      'verify-files': { type: 'boolean', default: false },
+      require: { type: 'string', default: '' },
+    },
+  });
+  const registryPath = path.resolve(values.registry!);
+  const artifactRoot = path.resolve(values['artifact-root']!);
+  const entries = parseRegistry(registryPath);
+  if (entries.length === 0) throw new Error('Artifact registry contains no entries.');
+  entries.forEach((entry, index) => validateEntry(entry, index + 1));
```
Comment (zouantchaw) [inline packages/scripts/src/dataset-factory/check-artifact-registry.ts]: [P2] Validate against the published Draft 2020-12 schema instead of this partial hand check. I changed one entry to `storage_class: "not_a_schema_value"` and added an unexpected top-level property; this command still returned `status: ok`. The current code also skips several required fields/enums, `additionalProperties`, uniqueness, digest scope/type coupling, and locator containment. Use a real schema validator with format support, add semantic file/directory/scope checks, and cover invalid registry fixtures.

File: packages/scripts/src/dataset-factory/smoke-v0.ts [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+    path.join(out('visual_family_graph_v0'), 'visual-family-graph-v0-report.json'),
+    path.join(out('research_enrichment_v0'), 'research-enrichment-v0-report.json'),
+    path.join(out('search_judgments_v0'), 'search-judgments-v0-report.json'),
+    path.join(out('reward_data_v0'), 'reward-data-v0-report.json'),
+  ]) {
+    assertJson(filePath);
+  }
+
+  const reportPath = path.join(outputRoot, 'dataset-factory-smoke-v0-report.json');
+  const outputTreeHashBeforeReport = treeDigest(outputRoot);
```
Comment (zouantchaw) [inline packages/scripts/src/dataset-factory/smoke-v0.ts:543]: [P2] Make this hash an assertion over deterministic output, not a volatile observation. Two identical fixture runs to the same path changed 49 files and produced different pre-report tree hashes because the invoked generators stamp `new Date()` into rows/reports; the harness also only requires each output to have at least one row. This can pass substantial behavior regressions and cannot prove repeatable hashes. Add an injectable/fixed clock for fixture mode and assert exact counts plus representative content or a committed expected tree hash.

**Review by zouantchaw — COMMENTED** [review]
Independent review verdict: changes needed before #65 can pass the durability gate (COMMENT review only).

Verification passed: canonical 27-artifact hash/count check; v0 smoke for all nine command families; scripts typecheck; 24 JSON/JSONL/NDJSON files with 81 line records parsed; diff check; actionable missing/corrupt artifact failures with exit 1; tracked-files-only git-archive run passed npm ci + registry + smoke + typecheck. Node v22.22.0 emitted the expected engine warning against >=23.5.0.

Four inline findings remain: reconstructibility of the registry graph/commands, one missing root command, incomplete schema enforcement, and nondeterministic/weak smoke assertions.

**Comment by zouantchaw** [comment]
Addressed independent review 4666485256 in commit 703b2ab.

1. Registry reconstruction: replaced the cyclic 27-entry graph with a 49-entry acyclic phase graph. Batch and benchmark trees are exact file sets; the 53-file batch is covered once across 13 phases. Four human-authored decision files are explicit immutable inputs, and their apply outputs use automated_with_human_inputs. The pre-model active-learning bootstrap is separate from the later model-informed refresh and was byte-verified against all five canonical label files.
2. Command wiring: added the root image-orientation:audit forwarder. Registry validation now checks every referenced root npm script exists.
3. Schema enforcement: the checker uses Ajv Draft 2020-12 plus format validation, then enforces digest/kind coupling, file-set membership, locator containment, storage coupling, stable-ID uniqueness, dependency/lineage existence, human-input boundaries, command existence, and cycle rejection. The self-test passes nine deterministic negative cases.
4. Deterministic smoke: Dataset Factory commands accept DATASET_FACTORY_FIXED_NOW, fixture input mtimes and reported mock API base are normalized, exact counts and representative rows are asserted, and the committed output-tree hash is d42c31f149637098a058637e3073a4a827674c0d1278df66ea870012a537145d. Two consecutive local runs and a separate git-archive run produced that hash.

Evidence: artifact self-test passed all nine cases; full canonical verification passed for 49 entries; scripts typecheck passed; 24 tracked JSON/JSONL/NDJSON files parsed (10 JSON, 103 JSONL/NDJSON records); diff/scope/secret/large-file audits passed. A tracked-files-only archive passed fresh npm ci, registry check, self-test, full smoke, and typecheck. Node v22.22.0 still emits the documented engine warning against >=23.5.0; external search snapshots and human decisions remain explicit non-automated lineage boundaries.

**Review by zouantchaw — COMMENTED** [review]
Independent re-review verdict: changes still needed before #65 passes. Two original findings are resolved (root orientation wiring and deterministic smoke); two remain partially open (registry reconstructibility and real filesystem containment).

Passing evidence: 49-entry schema/graph check; all nine built-in and independently invoked negative classes; canonical full hash/count/kind verification; acyclic 49-node graph; exact, non-overlapping 53/10/44-file batch and benchmark coverage; two identical smoke runs at d42c31f149637098a058637e3073a4a827674c0d1278df66ea870012a537145d with exact counters/content; scripts typecheck; 24 structured files/103 line records parsed; diff/scope/secret/large-file checks; and a true tracked-files-only archive after fresh npm ci. Node v22.22.0 still warns against >=23.5.0.

Four inline findings follow: two unreconstructible broad registry nodes, symlink containment, and timezone-less fixed-clock ambiguity.

File: docs/dataset-factory/artifact-registry.v0.jsonl [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
...ry reviewed/silver retrieval judgment bundle and expanded reranker artifacts.","source_artifact_ids":["dfv0_benchmark_quality_gold","dfv0_research_enrichment_v0","dfv0_visual_family_graph_v0"],"source_urls":["https://github.com/zouantchaw/mtl-archives-search/issues/65"]},"storage":{"storage_class":"local_filesystem","path_class":"repo_ignored_local","locator":"data/mtl_archives/reports/search_judgments_v0"},"generation":{"method":"automated","command":"npm run dataset-factory:search-judgments-v0","code_ref":"codex/65-dataset-factory-v0-durability","human_input_ids":[]},"dependency_ids":["dfv0_benchmark_quality_gold","dfv0_research_enrichment_v0","dfv0_visual_family_graph_v0"],"required_by":["dataset-factory:search-reranker-v0","dataset-factory:quality-search-guardrail-v0"],"rights_boundary":{"license_id":"cc-by-4.0-derived","attribution":"Archives de la Ville de Montreal; generated Dataset Factory analysis by MTL Archives.","commercial_use_allowed":true,"notes":"Derived metadata/evaluation artifacts inherit source attribution requirements; generated labels remain review artifacts, not source facts."},"created_at":"2026-06-30T08:16:37.170Z","creation_time_basis":"latest_file_mtime"}
```
Comment (zouantchaw) [inline docs/dataset-factory/artifact-registry.v0.jsonl]: [P1] Split the remaining search-judgment snapshot into honest phases. This entry registers 12 files, but `build-search-judgments-v0.ts` writes only the four root judgment/task/report files. The three `live_baseline_current` files and five `search_reranker_v0_expanded` files are pre-existing API-derived snapshots; this command neither creates them nor declares their API/manifest/label dependencies, while the entry is marked `automated`. A clean reconstruction therefore cannot reproduce this digest. Model the root four files as an automated file set and the baseline/reranker subtrees as separate `external_snapshot` file sets with exact commands and dependencies.

File: docs/dataset-factory/artifact-registry.v0.jsonl [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
...d","command":"npm run dataset-factory:quality-repair-v0 && npm run dataset-factory:quality-coverage-v0 && npm run dataset-factory:quality-derivatives-v0 && npm run dataset-factory:quality-failure-reconciliation-v0 && npm run dataset-factory:quality-rotation-sql-review-v0 && npm run dataset-factory:quality-search-guardrail-v0","code_ref":"codex/65-dataset-factory-v0-durability","human_input_ids":[]},"dependency_ids":["dfv0_manifest_clean","dfv0_image_quality_sample","dfv0_cleanup_embedding_bundle","dfv0_missing_images_rows","dfv0_missing_images_report","dfv0_image_orientation_audit","dfv0_image_orientation_sql","dfv0_image_artifact_audit","dfv0_image_artifact_decisions","dfv0_benchmark_quality_gold_search_baseline"],"required_by":["dataset-factory:production-apply-gate-v0"],"rights_boundary":{"license_id":"cc-by-4.0-derived","attribution":"Archives de la Ville de Montreal; generated Dataset Factory analysis by MTL Archives.","commercial_use_allowed":true,"notes":"Derived metadata/evaluation artifacts inherit source attribution requirements; generated labels remain review artifacts, not source facts."},"created_at":"2026-07-02T04:46:47.608Z","creation_time_basis":"latest_file_mtime"}
```
Comment (zouantchaw) [inline docs/dataset-factory/artifact-registry.v0.jsonl]: [P1] This command does not reconstruct the registered 636-file quality tree. The canonical snapshot contains more than twenty custom pilot/Kami/audit directories, while the command omits the `autoresearch:image-quality` runs and their custom flags; conversely, `quality-derivatives-v0` defaults to `audit_derivatives_v0`, which is not present in the registered tree. The hash currently verifies an existing snapshot, not the recorded generation command. Split this tree into phase/file-set or external-snapshot entries and record exact commands, inputs, clock values, and manual/external boundaries for each.

File: packages/scripts/src/dataset-factory/check-artifact-registry.ts [hunk 7]
```diff
@@ -0,0 +1,446 @@
...
+    stack.pop();
+    state.set(id, 'visited');
+  };
+  for (const id of byId.keys()) visit(id);
+  return byId;
+}
+
+function verifyArtifact(entry: RegistryEntry, artifactRoot: string): void {
+  if (entry.storage.path_class !== 'repo_ignored_local' && entry.storage.path_class !== 'tracked_fixture') return;
+  const artifactPath = path.resolve(artifactRoot, entry.storage.locator);
```
Comment (zouantchaw) [inline packages/scripts/src/dataset-factory/check-artifact-registry.ts:313]: [P2] Enforce containment after filesystem resolution. I created `artifactRoot/data/linked.jsonl` as a symlink to a sibling file outside `artifactRoot`, registered its correct digest/count, and `--verify-files` returned `status: ok`. The lexical `path.resolve`/`path.relative` check is bypassed because `fs.statSync` and `readFileSync` follow the link. Reject symlink locators/members with `lstat`, or resolve each existing target/member with `realpathSync` and assert the real path remains under the real artifact root; add this adversarial case to self-test coverage.

File: packages/scripts/src/dataset-factory/clock.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+const FIXED_NOW_ENV = 'DATASET_FACTORY_FIXED_NOW';
+
+export function datasetFactoryFixedNowIso(): string | null {
+  const fixedNow = process.env[FIXED_NOW_ENV]?.trim();
+  if (!fixedNow) return null;
+  const parsed = new Date(fixedNow);
```
Comment (zouantchaw) [inline packages/scripts/src/dataset-factory/clock.ts]: [P3] Require an explicit timezone in the fixed-clock value. `DATASET_FACTORY_FIXED_NOW=2026-07-09T12:00:00` is accepted, but normalizes to `12:00Z` under `TZ=UTC` and `16:00Z` under `America/Toronto`, so the same supposedly deterministic input can produce different artifact hashes. Validate a strict RFC 3339 timestamp ending in `Z` or an explicit numeric offset before constructing `Date`.

**Comment by zouantchaw** [comment]
Remediation for review 4666770734 and threads `PRRT_kwDOQFNJR86Pshmk`, `PRRT_kwDOQFNJR86Pshmw`, `PRRT_kwDOQFNJR86PtM_b`, `PRRT_kwDOQFNJR86PtM_f`, `PRRT_kwDOQFNJR86PtM_k`, and `PRRT_kwDOQFNJR86PtM_p` is pushed in `0254905`.

1. Search judgment snapshot: replaced the 12-file broad node with 3 exact phases covering 12/12 members once: a deterministic 4-file core, a 3-file immutable live-baseline snapshot, and a 5-file immutable expanded-reranker snapshot. API snapshots have parameter-complete commands and explicit mutable acquisition boundaries.
2. Quality repair snapshot: replaced the 636-file broad node with 25 exact phase entries covering 636/636 members once, with zero overlaps/misses/extras. Deterministic repair/coverage/guardrail phases are separate from remote-image, live-search, and interrupted-checkpoint snapshots. The stale absent `audit_derivatives_v0` command is removed; the empty `bounded_pixel_audit_500` workspace is explicitly excluded.
3. Symlink containment: verification now rejects symlinks in locator/member path components before content reads, enforces real-path containment, rejects nested tree symlinks, and checks cross-entry member uniqueness. Self-tests prove normal in-root acceptance plus leaf symlink, parent-directory symlink, and overlap rejection.
4. Clock: `DATASET_FACTORY_FIXED_NOW` now requires timezone-qualified RFC 3339 (`Z` or explicit offset), validates calendar/range fields, and normalizes equivalent Z/offset instants identically under UTC and America/Toronto. Date-only and timezone-less values reject.

Proof:
- registry/schema: 76 entries; schema check and 14-case self-test pass
- canonical `--verify-files`: pass against the populated read-only artifact root
- exact coverage: search 3 entries/12 files; quality 25 entries/636 files; no duplicate, missing, or extra members
- clock self-test: pass under both `TZ=UTC` and `TZ=America/Toronto`
- deterministic smoke: two consecutive runs both `d42c31f149637098a058637e3073a4a827674c0d1278df66ea870012a537145d` with exact expected counts/content
- scripts typecheck, 24 tracked Dataset Factory JSON/JSONL/NDJSON parse, and `git diff --check`: pass
- scope/security/size audit: 13 scoped files, no secret/signed-URL findings, no file over 1 MB
- tracked-files-only archive: fresh `npm ci`, registry check, self-test, clock test, full smoke, typecheck, and structured parse all pass

Threads remain unresolved for independent reviewer recheck.

**Review by zouantchaw — COMMENTED** [review]
Independent third-gate re-review at `0254905bb991051dd4e5115bbe064c2815274e55`: **issue #65's durability gate passes with no actionable findings**. The four findings from review 4666770734 are resolved, as are the two remaining original graph/schema threads.

Evidence: the 3 search entries cover exactly 12 canonical files once (4+3+5), and the 25 quality entries cover exactly 636 once, with no misses/extras/overlap. An independent walk verified all 76 nodes are dependency-complete and acyclic, all root commands exist, external/human boundaries are explicit, and every local artifact matches path/kind/count/row-count/SHA-256. The 14-case self-test passed; separate mutations rejected enum, extra-property, scope/kind, containment, duplicate, missing dependency, cycle, unknown command, and missing acquisition-boundary cases. Independent regular-file/leaf-symlink/parent-symlink checks passed/rejected as intended before hashing.

Clock tests passed under UTC and America/Toronto; Z/offset instants match, ambiguous forms reject, and an ordinary command used wall time. Two clean-archive smoke runs were byte-identical at `d42c31f149637098a058637e3073a4a827674c0d1278df66ea870012a537145d` with all exact counters/content. Fresh tracked-files-only `npm ci`, registry check/self-test, clock tests, smoke, scripts typecheck, structured parsing, diff, scope, secret/signed-URL, and size audits passed.

Residual limits: Node 22.22.0 warns against declared `>=23.5.0`; full hash verification needs the populated artifact root; mutable external snapshots are recaptures, not byte reproduction; fixture smoke is not live quality proof. The checker closes static symlink escapes but still has the usual local TOCTOU window between path validation and later reads if an actively hostile writer mutates the artifact tree concurrently.
